lib/default.nix

```diff
@@ -143,6 +143,15 @@ with haskellLib;
      (lib.filterAttrs (name: package: (package.isHaskell or false) && packageSel package) haskellPackages))
     // { recurseForDerivations = true; };
 
+  # Equivalent to collectComponents with (_: true) as selection function.
+  # Useful for pre-filtered package-set.
+  #
+  # For example:
+  #
+  #    myHaskellPackages = selectProjectPackages hsPkgs;
+  #    myTests = collectComponents' "tests" myHaskellPackages;
+  collectComponents' = group: collectComponents group (_: true);
```

Comment:
We use something like this in our own codebase as well, so I think this `collectComponents'` function is a nice addition:

```nix
our-exes = collectComponents "exes" (_: true) (selectLocalPackages self.hsPkgs);
```